human_cancer_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,778 @@
+
+# In[36]:
+
+
+# Load Powers et. al. results file
+powers_rank_filename = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    "Hallmarks_qvalues_GSEAPreranked.csv"
+)
+
+
+# In[37]:
+
+
+# Read Powers et. al. data
+# This file contains qvalue results for hallmark pathways across ~400 experiments
+powers_rank_df = pd.read_csv(powers_rank_filename, header=0, index_col=0)
+powers_rank_df.drop(['Category'], axis=1, inplace=True)
+print(powers_rank_df.shape)
+powers_rank_df.head()
+
+
+# In[38]:
+
+
+# Count the number of experiments where a given pathway was found to be enriched (qvalue < 0.05)
+total_num_experiments = powers_rank_df.shape[1]
+frac_enriched_pathways = ((powers_rank_df < 0.05).sum(axis=1) / total_num_experiments)
+
+# Rank pathways from 0-50, 50 indicating that the pathways was frequently enriched
+pathway_ranks = frac_enriched_pathways.rank()
+
+powers_rank_stats_df = pd.DataFrame(
+    data={
+        'Fraction enriched': frac_enriched_pathways.values,
+        'Powers Rank':pathway_ranks.values
+    },
+    index=powers_rank_df.index
+)
+powers_rank_stats_df.sort_values(by="Powers Rank", ascending=False).head()
+
+
+# In[39]:
+
+
+# Save reference file for input into comparison
+powers_rank_processed_filename = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    "Hallmarks_qvalues_GSEAPreranked_processed.tsv"
+)
+
+powers_rank_stats_df.to_csv(powers_rank_processed_filename, sep="\t", )
+
+
+# In[53]:
+
+
+figure_filename = f"pathway_ranking_{col_to_rank_pathways}.svg"
+
+process.compare_pathway_ranking(
+    summary_pathway_ranks,
+    powers_rank_processed_filename,
+    figure_filename
+)
+
+
+# * Our method ranked pathways using median adjusted p-value score across simulated experiments.
+# * Powers et. al. ranked pathways based on the fraction of experiments they had adjusted p-value < 0.05.
+# * Next let's try to use the fraction of adjusted p-value < 0.05 for our method and re-compare (below)
+# 
+# **Takeaway:**
+# * Previously we compared pathway ranks obtained from (recount2)-trained VAE model vs pathway ranking based on manual curation using Powers et. al. This [PR](https://github.com/ajlee21/generic-expression-patterns/blob/807377d76f63b6282c62255d7b160feb8585e0e2/human_analysis/2_identify_generic_genes_pathways.ipynb) shows that there was no correlation.
+# 
+# * Here we are comparing pathway ranks obtained from a (Powers et. al.)-trained VAE model vs pathway ranking based on manual curation using Powers et. al datasets. Based on this correlation plot there is a high correlation.
+
+# **Conclusion:**
+# 
+# * We can get relatively similar generic genes training our VAE model on a cancer-specific dataset (Powers et. al.) and testing on a genenal dataset (Crow et. al.). These generic genes are not *that* context-specific at the extremes
```

Comment:
Can you clarify what you mean by "similar" here? Do you mean the genes are similar between experiments, or between what you find using simulated data and what the paper found using real data?

I think you mean the latter (comparing the scatter plot above to the one in `human_analysis/2_identify_generic_genes_pathways.py`, it looks like your results here are much more correlated with the ones from Powers et al, which I'm guessing is what you're saying), but would be good to make this clear.

Reply:
Good point. I will clarify this